packages/lesswrong/components/comments/CommentsItem/CommentBody.tsx

```diff
@@ -87,23 +112,49 @@ const CommentBody = ({
     highlights = votingSystem.getCommentHighlights({comment, voteProps});
   }
 
-  const contentBody = <ContentStyles contentType={contentType} className={classes.root}>
-    <ContentItemBody
-      ref={commentBodyRef ?? undefined}
-      className={bodyClasses}
-      dangerouslySetInnerHTML={{__html: innerHtml }}
-      description={`comment ${comment._id}`}
-      nofollow={(comment.user?.karma || 0) < nofollowKarmaThreshold.get()}
-      replacedSubstrings={highlights}
-    />
-  </ContentStyles>
+  const renderContent = (content: string) => (
```

Comment:
Prefer not to use the inline closure pattern for dynamically rendering components inside of other components, since it breaks React's ability to avoid re-rendering any child components that didn't change as a result of this component's state changes.

(In this case you may need to create a couple wrapper components if you want to avoid a bunch of duplicate code.)